server/mediaserver.go

```diff
@@ -322,6 +353,11 @@ func endRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 		s.LivepeerNode.VideoNetwork.UpdateMasterPlaylist(manifestID, nil)
 		//Remove the stream from cache
 		s.LivepeerNode.VideoCache.EvictHLSStream(core.StreamID(hlsID))
+
+		if s.LivepeerNode.MonitorMetrics {
+			monitor.LogStreamEndedEvent(manifestID, s.VideoNonce[rtmpStrm.GetStreamID()])
```

Comment:
We probably need to check if `s.VideoNonce[rtmpStrm.GetStreamID()]` exists, since there are a couple situations where it could still be unset: 

If the `if len(s.rtmpStreams) > 0`  check fails, the `gotRTMPStreamHandler` function will exit before the VideoNonce is set.

If `createRTMPStreamIDHandler` errors out (and returns an empty string), then `gotRTMPStreamHandler` will never be called, but `endRTMPStreamHandler` will still be called.

Also, should this also be locked by the mutex?